CommandLineOperations/RepetitionCounter.cs

```diff
@@ -0,0 +1,181 @@
+                {
+                    case string sourceArg when sourceArg.Contains(Argument.source.ToValidArgument()):
+                        sourceFilePathTask = Task.Factory.StartNew(() => GetSourceFilePath(sourceArg));
+                        break;
+                    case string targetArg when targetArg.Contains(Argument.target.ToValidArgument()):
+                        targetJsonFileTask = Task.Factory.StartNew(() => GetTargetJsonFile(targetArg));
+                        break;
+                    case string repetitionArg when repetitionArg.Contains(Argument.repetitions.ToValidArgument()):
+                        break;
+                    default:
+                        if (!allowedArguments.Any(arg => argument.Contains(arg.ToValidArgument())))
+                            throw new ArgumentException($"Wrong {argument} argument");
+                        break;
+                }
+            }
+
+            if (targetJsonFileTask != null)
+            {
+                allTasks = new Task[] { sourceFilePathTask, targetJsonFileTask };
+
+                Task.WaitAll(allTasks);
+
+                targetJsonPath = targetJsonFileTask.Result;
+            }
+            else
+            {
+                sourceFilePathTask.Wait();
+            }
+
+            sourceFilePath = sourceFilePathTask.Result;
+
+            textFromSourceFile = await CommandLineReader.GetTextFromSourceFileAsync(sourceFilePath, args);
+
+            Dictionary<string, int> keyValueOfRepetitiveWords = GetDictionaryOfRepetativeWords(textFromSourceFile);
+
+            WriteDictionaryToJsonFileInDescending(keyValueOfRepetitiveWords, targetJsonPath, args);
+        }
+
+        private Dictionary<string, int> GetDictionaryOfRepetativeWords(string textFromSourceFile)
+        {
+            char[] delimiterChars = { ' ', ',', '.', ':', ';', '!', '?', '\t', '\r', '\n', '—', '-', '"' };
+
+            Dictionary<string, int> dictionaryWithRepetitiveWords = new Dictionary<string, int>();
+
+            List<string> listOfAllWords = textFromSourceFile.ToLower().Split(delimiterChars, StringSplitOptions.RemoveEmptyEntries).ToList();
+
+            listOfAllWords.ForEach(word =>
+            {
+                if (dictionaryWithRepetitiveWords.ContainsKey(word))
+                {
+                    dictionaryWithRepetitiveWords[word]++;
+                }
+                else
+                {
+                    dictionaryWithRepetitiveWords.Add(word, 1);
+                }
+            });
+
+            return dictionaryWithRepetitiveWords;
+        }
+
+        private void WriteDictionaryToJsonFileInDescending(Dictionary<string, int> dictionaryWithRepetitiveWords, string targetJsonPath, string[] args)
+        {
+            List<string> orderedTextInJsonFormat = dictionaryWithRepetitiveWords.OrderByDescending(word => word.Value).Select(keyValuePair => string.Format("{0}:{1}", keyValuePair.Key, keyValuePair.Value)).ToList();
```

Comment:
try to move diffent linq methods into different lines if the initial line is too long